Library/Homebrew/unpack_strategy.rb

```diff
@@ -88,24 +132,32 @@ def self.detect(path, prioritize_extension: false, type: nil, ref_type: nil, ref
     strategy.new(path, ref_type:, ref:, merge_xattrs:)
   end
 
-  attr_reader :path, :merge_xattrs
+  sig { returns(Pathname) }
+  attr_reader :path
+
+  sig { returns(T.nilable(T::Boolean)) }
+  attr_reader :merge_xattrs
 
+  sig {
+    params(path: T.any(String, Pathname), ref_type: T.nilable(String), ref: T.nilable(String),
+           merge_xattrs: T.nilable(T::Boolean)).void
```

Comment:
```suggestion
           merge_xattrs: T::Boolean).void
```

unless it's a tri-state thing

Reply:
This is because `merge_xattrs: nil`.

Reply:
Yup. That should be changed unless it has to be `nil` and can't be `false` instead.